relayer/src/upgrade_chain.rs

```diff
@@ -94,11 +97,16 @@ pub fn build_and_send_ibc_upgrade_proposal(
         }),
     };
 
-    let mut buf_proposal = Vec::new();
-    prost::Message::encode(&proposal, &mut buf_proposal).unwrap();
+    let proposal = if !opts.legacy {
+        Proposal::Default(proposal)
+    } else {
+        Proposal::Legacy(proposal.into())
+    };
```

Comment:
Nit: it may be easier to read if we use the plain test for `legacy` here, instead of the negated test.
```suggestion
    let proposal = if opts.legacy {
        Proposal::Legacy(proposal.into())
    } else {
        Proposal::Default(proposal)
    };
```